default.nix

```diff
@@ -1,76 +1,79 @@
-{ nixpkgs ? import <nixpkgs> {} }:
-with nixpkgs; let
-  agdaDeps = with (import (builtins.fetchTarball
+{
+  pkgs ? import <nixpkgs> {},
+}:
+
+with pkgs;
+let
+  customAgda = (import (builtins.fetchTarball
     https://github.com/nixos/nixpkgs/tarball/a7ecde854aee5c4c7cd6177f54a99d2c1ff28a31) ({ # 21.11
       overlays = [ (import (fetchFromGitHub {
         repo = "agda";
         owner = "input-output-hk";
         rev = "d5ea03b96328f38741efef4535197076ff0e05d5";
         sha256 = "WiadZWZvPWcR49JkiXPiMKW3plRjBlR94wyg/aoEoG8=";
       })).overlay ];
-    } // (if pkgs.system == "aarch64-darwin" then { system = "x86_64-darwin"; } else {}))); let
```

Comment:
I removed the nested `let` - it seemed more difficult to follow and require a few passing on the variables between the scopes

Reply:
Your version is definitely clearer than mine!